salt/minion.py

```diff
@@ -920,6 +944,7 @@ def __init__(self, opts, context=None):
         # Late setup of the opts grains, so we can log from the grains module
         import salt.loader
 
+        _sync_grains(opts)
```

Comment:
Only slightly, since it has to do the sync and an extra write to the /var/cache/salt for the grains and extmods area, but it is very slight, that is, any write to disk is slowing but not noticeably so